network-api/networkapi/wagtailpages/pagemodels/customblocks/listing.py

```diff
@@ -13,6 +16,22 @@ class ListingCard(blocks.StructBlock):
 
     link = blocks.PageChooserBlock(required=False, help_text="Page that this should link out to.")
 
+    link_url = blocks.CharBlock(
+        required=False,
+        help_text="Optional URL that the header should link out to.",
+    )
+
+    def clean(self, value):
+        result = super().clean(value)
+        errors = {}
+
+        if value["link"] and value["link_url"]:
+            errors["link"] = ErrorList(["Please choose between a link OR a URL value."])
+        if errors:
+            raise StructBlockValidationError(errors)
```

Comment:
I guess the same question as in #9739, about why we are using the `errors` instead of raising the error directly.

Reply:
Yep, reflected that in 470041f6164fa3dc18fb7a33ad8612ca82daae96